pkg/diagnostics/types/flags.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"github.com/openshift/origin/pkg/cmd/server/start"
+	flag "github.com/spf13/pflag"
+	"strings"
+)
+
+type Flags struct {
+	Diagnostics    List // named diagnostics to run
+	OpenshiftPath  string
+	OscPath        string
+	LogLevel       int
+	Format         string          // format of output - text/json/yaml
+	OpenshiftFlags *flag.FlagSet   // flags that were set on the command run
```

Comment:
Try to keep these logically distinct.  We've been splitting commands into stages:
 1. Gather the data required for the command into an intermediate (`Options`) object.  This is done via binding or in `.Complete`
 1. Validate the data given for a command. `.Validate`
 1. Run the guts of the command based on the `Options`. 

This approach makes it easier to write unit tests and makes it possible to delegate from one command to another with minimal pain.  As examples, you can see our usage in `kubectl config` (testing), `osc login` and `osc project` (delegatees), and `openshift start` (delegator).  We also re-use these commands in integration testing, so there's more examples there.